lib/ui/platform_dispatcher.dart

```diff
@@ -308,6 +308,63 @@ class PlatformDispatcher {
     _invoke(onMetricsChanged, _onMetricsChangedZone);
   }
 
+  /// A callback invoked when a given platform focus change affects flutter.
+  ///
+  /// The framework should set the callback to subscribe to those events:
+  ///
+  /// If the platform focus moves from any element to a view with id 2,
+  /// in a forward direction, the engine should peform the following call:
+  ///
+  /// PlatformDispatcher.instance.onPlatformFocusChange(
+  ///   ViewFocusEvent(2, ViewFocusState.forwardFocused),
+  /// );
+  ///
+  /// If the platform focus moves from that view with id 2, to a view with id 3,
+  /// in a backwards direction, the engine should peform the following call:
+  ///
+  /// PlatformDispatcher.instance.onPlatformFocusChange(
+  ///   ViewFocusEvent(3, ViewFocusState.backwardsFocused),
+  /// );
+  ///
+  /// If the platform focus moves from that view with id 3, to a non flutter view,
+  /// the engine should peform the following call:
+  ///
+  /// PlatformDispatcher.instance.onPlatformFocusChange(
+  ///   ViewFocusEvent(3, ViewFocusState.unfocused),
+  /// );
+  ///
+  /// If the platform focus moves any element to a flutter view with id 2, in a
+  /// undirected form, the engine should perform the following call:
+  ///
+  /// PlatformDispatcher.instance.onPlatformFocusChange(
+  ///   ViewFocusEvent(2, ViewFocusState.focused),
+  /// );
+  ViewFocusChangeCallback? get onPlatformFocusChange => _onPlatformFocusChange;
+  ViewFocusChangeCallback? _onPlatformFocusChange;
+  Zone _onPlatformFocusChangeZone = Zone.root;
+  set onPlatformFocusChange(ViewFocusChangeCallback callback) {
+    _onPlatformFocusChange = callback;
+    _onPlatformFocusChangeZone = Zone.current;
+  }
+
+  /// Calling this method tells the engine to change the platform focus.
```

Comment:
nit: this doc comment could contain some linked breadcrumbs to things like [FlutterView] and [ViewFocusSate].